ml-agents-envs/mlagents/envs/environment.py

```diff
@@ -583,7 +592,16 @@ def _close(self):
         self._loaded = False
         self.communicator.close()
         if self.proc1 is not None:
-            self.proc1.kill()
+            # Wait a bit for the process to shutdown, but kill it if it takes too long
+            try:
+                self.proc1.wait(timeout=self.timeout_wait)
```

Comment:
I think this addresses the main concern I had, which was a user would have to manually close 15 unity processes.  But it looks like you are waiting here to see if they play nice, then drop the hammer if not.

Reply:
True, it'll still eventually get killed. But we're relying on the Academy interpreting the commands and closing the application; a game developer might expect the signals to get propagated.

Maybe we should make Academy listen for signals instead?

Reply:
ehhh, I don't like the sound of that.

Reply:
OK, punting on this for now. If it comes up, we'll add a flag to toggle `start_new_session`